Assets/Scripts/Enemy/EnemyController.cs

```diff
@@ -0,0 +1,78 @@
+/*
+ * EnemyController.cs
+ * Authors: Jehdi Aizon, Josh Coss
+ * Created: January 17, 2024
+ * Description: Handles base enemy controller movements.
+ */
+
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+/// <summary>
+/// Controls the basic behavior of an enemy, such as movement and state transitions.
+/// </summary>
+public class EnemyController : MonoBehaviour
+{
+    [Header("Enemy Scriptable Object")]
+    [Tooltip("EnemyScriptableObject for initial enemy stats")]
+    [SerializeField]
+    private EnemyScriptableObject enemyStats;
+
+    private EnemyStateMachine enemyStateMachine;
+    [HideInInspector] public Rigidbody2D rb;
+
+    // Movement speed of the enemy
+    protected float movementSpeed;
+
+    // Attack speed measured in damage per second
+    private float attackSpeed;
+
+    // Strength: amount of damage inflicted by the enemy
+    private float strength;
+
+    // Health points of the enemy
+    private float healthPoints;
+
+    // Status effect of the enemy (placeholder enum, could be expanded)
+    private enum EnemyStatus { };
```

Comment:
We could probably already define the enums for Enemy Status here: Stunned, Immobilized, Poisoned and Normal (refer to documentation)

Reply:
Added